### PR TITLE
Replace fail+catch pattern with assertThatThrownBy in trino-blackhole

### PR DESCRIPTION
**Pull request by choiwaiyiu**
Relates to https://github.com/trinodb/trino/issues/6553

### CODE REVIEW

File: plugin/trino-blackhole/src/test/java/io/trino/plugin/blackhole/TestBlackHoleSmoke.java [hunk 1]
```diff
@@ -117,18 +116,14 @@ public void notAllPropertiesSetForDataGeneration()
...
-            // expected exception
-        }
+        assertThatThrownBy(() -> assertThatQueryReturnsValue(
+                format("CREATE TABLE nation WITH ( %s = 3, %s = 1 ) as SELECT * FROM tpch.tiny.nation",
+                        ROWS_PER_PAGE_PROPERTY,
+                        SPLIT_COUNT_PROPERTY),
+                25L,
+                session))
+                .isInstanceOf(RuntimeException.class)
+                .hasMessage("All properties [split_count, pages_per_split, rows_per_page] must be set if any are set");
```
Comment (Praveen2112) [inline plugin/trino-blackhole/src/test/java/io/trino/plugin/blackhole/TestBlackHoleSmoke.java:126]: Isn't error message is a bit incomplete ?

Comment (Praveen2112) [reply plugin/trino-blackhole/src/test/java/io/trino/plugin/blackhole/TestBlackHoleSmoke.java:126]: Kind of nit

**Review by Praveen2112 — APPROVED** [review]

File: plugin/trino-blackhole/src/test/java/io/trino/plugin/blackhole/TestBlackHoleSmoke.java [hunk 2]
```diff
@@ -117,18 +116,14 @@ public void notAllPropertiesSetForDataGeneration()
...
-                            ROWS_PER_PAGE_PROPERTY,
-                            SPLIT_COUNT_PROPERTY),
-                    25L,
-                    session);
-            fail("Expected exception to be thrown here!");
-        }
-        catch (RuntimeException ex) {
-            // expected exception
-        }
+        assertThatThrownBy(() -> assertThatQueryReturnsValue(
```
Comment (findepi) [inline plugin/trino-blackhole/src/test/java/io/trino/plugin/blackhole/TestBlackHoleSmoke.java:119]: replace `assertThatQueryReturnsValue` with eg `queryRunner.execute` -- the intent here is _not_ to run an assertion on the results.

**Review by ebyhr — APPROVED** [review]

**Comment by ebyhr** [comment]
Merged, thanks!

**Pull request merged**